### PR TITLE
Include `scannedDocuments` and `ocrData` population for transformation request object

### PR DESCRIPTION
**Pull request by doncem**
### JIRA link (if applicable) ###

[Create Case: pre-validation of callback request](https://tools.hmcts.net/jira/browse/BPS-738)

### Change description ###

Final step in validating / converting `CaseDetails` object to `ExceptionRecord` so it can be pushed to transformation API. Please follow the documentation links provided in #560 description.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CallbackValidations.java [hunk 1]
```diff
@@ -202,4 +203,18 @@ static boolean hasOcr(CaseDetails theCase) {
             .map(o -> Validation.<String, Instant>valid(Instant.parse((String) o)))
             .orElse(invalid("Missing " + dateField));
     }
+
+    /**
+     * Used in createCase callback only. Can be reviewed later and with BPS-746 - revised the usage.
+     * @param theCase from CCD.
+     * @return Validation of it which is always valid
```
Comment (nitinprabhu) [inline src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CallbackValidations.java]: in the callback I don't remember ccd sends an empty list or null value if the OCR data is not present. Should we not fail of scanOCRData is empty or null or there would be nothing to be transformed later

File: src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CreateCaseCallbackService.java [hunk 2]
```diff
@@ -58,17 +65,53 @@ public CreateCaseCallbackService() {
...
+    private Validation<String, List<ScannedDocument>> getScannedDocuments(CaseDetails caseDetails) {
+        return Try.of(() ->
+            Optional.ofNullable(caseDetails)
+                .map(Documents::getScannedDocuments)
+                .orElse(emptyList())
+                .stream()
+                .map(items -> items.get("value"))
+                .filter(item -> item instanceof Map)
+                .map(item -> (Map<String, Object>) item)
+                .map(document -> new ScannedDocument(
```
Comment (nitinprabhu) [inline src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CreateCaseCallbackService.java]: can we please extract it to a new method ?

File: src/test/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CreateCaseCallbackServiceTest.java [hunk 3]
```diff
@@ -90,4 +93,57 @@ void should_warn_about_missing_classification() {
...
+                "Invalid scannedDocuments format. Error: No enum constant " + DocumentType.class.getName() + ".OTHERS"
+            );
+        String match =
+            "Invalid OCR data format. Error: (class )?java.lang.Integer cannot be cast to (class )?java.lang.String.*";
+        assertThat(output.getLeft())
+            .filteredOn(string -> string.startsWith("Invalid OCR data format."))
+            .hasSize(1)
+            .element(0)
+            .asString()
+            .matches(match);
```
Comment (nitinprabhu) [inline src/test/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CreateCaseCallbackServiceTest.java]: can put the string value here directly ?

File: src/test/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CreateCaseCallbackServiceTest.java [hunk 4]
```diff
@@ -90,4 +93,57 @@ void should_warn_about_missing_classification() {
         assertThat(output.isLeft()).isTrue();
         assertThat(output.getLeft()).containsOnly("Missing journeyClassification");
     }
+
+    @Test
+    void should_report_errors_when_data_is_in_invalid_form() {
```
Comment (nitinprabhu) [inline src/test/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CreateCaseCallbackServiceTest.java]: should we have two case one when ocr is invalid and other when we can't let's say parse scanned docs ?

File: src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CallbackValidations.java (hunk 1 above)
Comment (doncem) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CallbackValidations.java]: Fair point. OK, will restrict this to have ocr strictly

File: src/test/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CreateCaseCallbackServiceTest.java (hunk 4 above)
Comment (doncem) [reply src/test/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CreateCaseCallbackServiceTest.java]: Could be just thought it's fairly bulky to set up requirements and test things. Later on 2 more components are coming in to play. Quite possible entire validation code can be extracted to component the way it is now so it doesn't end up like _attach to case_ callback service

tl;dr - will split for clarity

File: src/test/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CreateCaseCallbackServiceTest.java (hunk 3 above)
Comment (doncem) [reply src/test/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CreateCaseCallbackServiceTest.java]: the line exceeds 120 limit ;)
